hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoTransferValidator.java

```diff
@@ -113,7 +113,8 @@ public void pureChecks(@NonNull final CryptoTransferTransactionBody op) throws P
             validateFalsePreCheck(nftIds.size() < nftTransfers.size(), TOKEN_ID_REPEATED_IN_TOKEN_LIST);
 
             // Verify that one and only one of the two types of transfers (fungible or non-fungible) is present
-            validateFalsePreCheck(!nonZeroFungibleValueFound && nftIds.isEmpty(), EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS);
+            validateFalsePreCheck(
+                    uniqueTokenAcctIds.isEmpty() && nftIds.isEmpty(), EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS);
```

Comment:
Don't we need both of these checks? It's ok for `uniqueTokenAcctIds` and `nftIds` to be empty if there's a non-zero fungible value found

Reply:
I'm not sure about that. And I don't see such a check in mono(PureTransferSemanticChecks.checkTokenTransfersList). But I might be missing something. @Neeharika-Sompalli thoughts?

Reply:
yes @mhess-swl @vtronkov . We are okay if they are having zero amount. But both should not be set. So, agree with this change to be valid.